src/components/common/ChainSwitcher/index.tsx

```diff
@@ -15,7 +16,17 @@ const ChainSwitcher = ({ fullWidth }: { fullWidth?: boolean }): ReactElement | n
   const handleChainSwitch = useCallback(() => {
     if (!chain) return
     const chainId = hexValue(parseInt(chain.chainId))
-    onboard?.setChain({ chainId })
+
+    const wallets = onboard?.state.get()?.wallets || []
+    const wallet = getConnectedWallet(wallets)
+
+    if (wallet && isHardwareWallet(wallet)) {
+      onboard?.disconnectWallet({ label: wallet.label }).then(() => {
+        connectWallet(onboard, { autoSelect: wallet.label })
```

Comment:
Note: the address scanner modal opens on Ethereum by default and you must manually select the current network. I will investigate setting this (it would need to be implemented in onboard, unless we reinitialise onboard with 1 chain configuration before connecting a hardware wallet).

![image](https://user-images.githubusercontent.com/20442784/200343990-e1cc1f46-290d-4ab6-8777-0ad41f05d379.png)